Sources/NIO/Linux.swift

```diff
@@ -132,5 +136,60 @@ internal enum Linux {
         }
         return fd
     }
+
+    private static func contentsOfFile(path: String) throws -> Substring {
+        let fh = try NIOFileHandle(path: path)
+        defer { try? fh.close() }
+        // linux doesn't properly report /sys/fs/cgroup/* files lengths so we use a reasonable limit
+        let buffSize = 1024
+        var buf = ByteBufferAllocator().buffer(capacity: buffSize)
+        try buf.writeWithUnsafeMutableBytes(minimumWritableBytes: buffSize) { ptr in
+            let res = try fh.withUnsafeFileDescriptor { fd -> IOResult<ssize_t> in
+                return try Posix.read(descriptor: fd, pointer: ptr.baseAddress!, size: buffSize)
+            }
+            switch res {
+            case .processed(let n):
+                return n
+            case .wouldBlock:
+                throw LinuxError()
+            }
+        }
+        guard let content = buf.readString(length: buf.readableBytes) else { throw LinuxError() }
+        return content.prefix(while: { $0 != "\n" })
+    }
+
+    private static func toInt<S: StringProtocol>(_ s: S) -> Int? {
+        return Int(s)
+    }
+
+    private static func countCoreIds(cores: Substring) throws -> Int {
+        let ids = cores.split(separator: "-")
+        guard
+            let first = ids.first.flatMap(toInt),
```

Comment:
instead of `toInt`, you can just write `ids.first.flatMap { Int($0) }` or possibly `ids.first.flatMap(Int.init)`

Reply:
@weissi this is what I was initially doing but it doesn't work with `Substring`. I first went with `ids.first.map(String.init).flatMap(Int.init)` but then I thought the `toInt` alternative would look better. I'm fine either way so let me know if you prefer the other way. :)

Reply:
I went with `ids.first.map(String.init).flatMap(Int.init)` again :)